independent-projects/qute/core/src/test/java/io/quarkus/qute/UserTagTest.java

```diff
@@ -17,17 +17,17 @@ public void testUserTag() {
 
         Template tag = engine.parse("{#if showImage}{it.name}{#else}nope{/if}");
         engine.putTemplate("my-tag-id", tag);
-        Template template1 = engine.parse("{#myTag order showImage=true /}");
-        Template template2 = engine.parse("{#myTag order /}");
-        Template template3 = engine.parse("{#myTag showImage=false /}");
-        Template template4 = engine.parse("{#each this}{#myTag showImage=false /}{/each}");
 
         Map<String, Object> order = new HashMap<>();
         order.put("name", "Herbert");
-        assertEquals("Herbert", template1.render(Collections.singletonMap("order", order)));
-        assertEquals("nope", template2.render(Collections.singletonMap("order", order)));
-        assertEquals("nope", template3.render(Collections.singletonMap("order", order)));
-        assertEquals("nope", template4.render(Collections.singletonMap("order", order)));
+        assertEquals("Herbert",
+                engine.parse("{#myTag order showImage=true /}").render(Collections.singletonMap("order", order)));
+        assertEquals("nope", engine.parse("{#myTag order /}").render(Collections.singletonMap("order", order)));
+        assertEquals("nope", engine.parse("{#myTag showImage=false /}").render(Collections.singletonMap("order", order)));
+        assertEquals("nope",
+                engine.parse("{#each this}{#myTag showImage=false /}{/each}").render(Collections.singletonMap("order", order)));
```

Comment:
As I see tests mentioning `this` I have to ask what it refers to?

Reply:
In this particular case it refers to the data object passed to the template. `this` always returns the last context object on the stack.

Reply:
If it's not documented, it should not be implemented. If it's implemented it should be documented.
ATM I don't see the value of having `this` be implemented and resolved, so I suggest we disable it.